docs/tutorials/qsimcirq_gcp.md

```diff
@@ -0,0 +1,379 @@
+
+
+
+![alt_text](../images/image5.png )
+
+
+You will have a popup window connecting you to the simulation VM. You should have a command prompt:
+
+
+
+
+
+![alt_text](../images/image6.png )
+
+
+Start the container:
+
+
+```console
+    $ docker run -v `pwd`:/homedir -p 8888:8888 gcr.io/quantum-builds/github.com/quantumlib/jupyter_qsim:latest
+```
+
+
+The output will be something like:
+
+
+```console
+ Unable to find image 'gcr.io/quantum-291919/jupyter_qsim:latest' locally
+ latest: Pulling from quantum-291919/jupyter_qsim
+ 3c72a8ed6814: Pull complete 
+ 2bd71a698eae: Pull complete 
+ Digest: sha256:d0d0040c6ef9925719459e736631c9fec957b94d377009c591d6285143ebb626
+ ...    
+     To access the notebook, open this file in a browser:
+         file:///root/.local/share/jupyter/runtime/nbserver-1-open.html
+     Or copy and paste one of these URLs:
+         http://79804d33f250:8888/?token=d7a53e728e3dff08c6ed12a15810471486b4c15828a3d70a
+      or http://127.0.0.1:8888/?token=d7a53e728e3dff08c6ed12a15810471486b4c15828a3d70a
+```
+
+
+Copy the last URL (it starts with `127.0.0.1`). Replace `127.0.0.1` with the IP address of the VM 
+
+
+[created above](#build-a-container-optimized-vm): 
+
+
+```console
+ http://[EXTERNAL_IP_ADDRESS]:8888/?token=7191178ae9aa4ebe1698b07bb67dea1d289cfd0e0b960373
+```
+
+
+Paste that in your browser. You should now see the Jupyter UI:
+
+
+
+
+
+![alt_text](../images/image2.png )
+
+
+Navigate to  qsim > docs > tutorials. You will see:
+
+
+
+
+
+![alt_text](../images/image12.png )
+
+
+Click on qsimcirq.ipynb. This will load the notebook.
+
+You can skip the setup and go straight to the cell labeled "Full state-vector simulation":
+
+
+
+![alt_text](../images/image1.png )
+
+
+If you choose to modify the notebook, you can save it on the *qsim-1* VM from *File* -> *Save As*, and saving to `/homedir/mynotebook.ipynb`.  This will save in your home directory on your VM, if your directory is open to write.
+
+
+
+
+
+![alt_text](../images/image4.png )
+
+
+
+## Run interactively
+
+To run interactively within the container, you can open a second shell window to the VM as you did [above](#open-a-ssh-window-to-your-vm).
+
+Now, find the container ID with docker ps:
+
+
+```
+$ docker ps
+```
+
+
+
+![alt_text](../images/image11.png )
+
+
+The `CONTAINER_ID` is a UID something like "79804d33f250". Now you can connect to the container:
+
+
+```console
+    $ docker exec -it [CONTAINER_ID] /bin/bash
+```
+
+
+This shell is running inside the container, with Python3, Cirq and qsim installed. Start python3:
+
+
+```console
+	$ python3
+```
+
+
+You will see the output:
+
+
+```python
+    Python 3.6.8 (default, Apr 16 2020, 01:36:27) 
+    [GCC 8.3.1 20191121 (Red Hat 8.3.1-5)] on linux
+    Type "help", "copyright", "credits" or "license" for more information.
+    >>>
+```
+
+
+You are now ready for an interactive session running qsim.
+
```

Comment:
```suggestion
```console
    $ docker exec -it [CONTAINER_ID] /bin/bash
```

This shell is running inside the container, with Python3, Cirq and qsim installed. Start python3:

```console
	$ python3
```

You will see the output:

```python
    Python 3.6.8 (default, Apr 16 2020, 01:36:27) 
    [GCC 8.3.1 20191121 (Red Hat 8.3.1-5)] on linux
    Type "help", "copyright", "credits" or "license" for more information.
    >>>
```

You are now ready for an interactive session running qsim.
```